src/getAvailable.js

```diff
@@ -48,6 +54,16 @@ const getAvailable = async (req, res) => {
   });
 };
 
+async function collectAllAvailableGroups(user) {
+  const source = new (metaSources.get("groups"))(user);
```

Comment:
Does this mean we create a new instance of the metasource every time `/charon/getAvailable?prefix=/groups` is requested? Would that then restart the scheduling of `updateKnownDatasetsAndNarratives` if the `Groups` IIFE is executed each time?

clarification: I guess the IIFE part only is executed when the expression is parsed at server start time, but my question stands about whether `updateKnownDatasetsAndNarratives` is rerun if we are creating a new instance of the `Groups` metasource for every request.

Reply:
Good question - the IIFE runs only once (when the server starts) and since it's an anonymous function it's impossible to call it again (insofar as as I understand how JS referencing works). This function will initialise `datasetsPerGroup` etc,  call `updateKnownDatasetsAndNarratives` and scheduling it to run every ~6 hours via the `setTimeout`. 

The returned value (`MetaGroupSource`) is closed over the variables `datasetsPerGroup` etc. When we create an instance of this source, as we do each time a `/charon/getAvailable?prefix=/groups` request is handled, the constructor (etc) will run, but we won't run the anonymous function and therefore won't rerun `updateKnownDatasetsAndNarratives` etc.

(As an aside, the `MetaGroupSource` has access to the `updateKnownDatasetsAndNarratives` function, so it _could_ call it, but it doesn't. This is a common way to create private variables.)

I find adding a few `console.log` statements is a good way to test out understanding here (as well as reading docs etc of course!).